apps/etl-func/avro/message.json

```
@@ -0,0 +1,88 @@
+{
+  "name": "message",
+  "type": "record",
+  "namespace": "dto",
+  "doc": "Kafka JS schema for cosmos api container 'messages'",
+  "fields": [
+    {
+      "name": "schema_version",
+      "type": "int"
+    },
+    {
+      "name": "op",
+      "type": {
+        "type": "enum",
+        "name": "MessageCrudOperation",
+        "symbols": ["CREATE"]
+      }
+    },
+    {
+      "name": "id",
+      "type": "string"
+    },
+    {
+      "name": "sender_service_id",
+      "type": "string"
+    },
+    {
+      "name": "sender_user_id",
+      "type": "string"
+    },
+    {
+      "name": "is_pending",
+      "type": "boolean",
+      "default": true
+    },
+    {
+      "name": "content_subject",
+      "type": "string"
+    },
+    {
+      "name": "content_type",
+      "type": [
+        "null",
+        {
+          "type": "enum",
+          "name": "MessageContentType",
+          "symbols": ["GENERIC", "PAYMENT", "EU_COVID_CERT", "LEGAL", "PN"]
+        }
+      ],
+      "default": null
+    },
+    {
+      "name": "content_payment_data_amount",
+      "type": "int",
+      "default": 0
+    },
+    {
+      "name": "content_payment_data_notice_number",
+      "type": "string",
+      "default": "undefined"
```

Why `"undefined"`?

---

At this level we can't discriminate if the message is a payment message so this field should be optional

---

so the default value should be `null` (?)

```suggestion
      "name": "content_payment_data_notice_number",
      "type": ["null", "string"],
      "default": null
```

---

https://github.com/pagopa/io-messages/pull/40/commits/cd2ada3bc076f086566273d584282a97d491c7ec